install_linux.sh

```diff
@@ -174,19 +197,33 @@ sudo chown root:root $PWD/tools/cluster/cleanup.sh
 sudo chmod 755 $PWD/tools/cluster/cleanup.sh
 sudo chmod +x $PWD/tools/cluster/cleanup.sh
 
-print_message "Installing IPMI tools"
-if lsb_release -is | grep -q "Fedora"; then
-    sudo dnf -y install ipmitool
-else
-    sudo apt-get install -y freeipmi-tools ipmitool
-fi
 
+if [[ $no_msr_tools != true ]] ; then
+    print_message "Installing msr-tools"
+    print_message "Important: If this step fails it means msr-tools is not available on you system"
+    print_message "If you do not plan to use RAPL you can skip the installation by appending '-r'"
+    if lsb_release -is | grep -q "Fedora"; then
+        sudo dnf -y install msr-tools
+    else
+        sudo apt-get install -y msr-tools
+    fi
+fi
 
-print_message "Adding IPMI to sudoers file"
-echo "ALL ALL=(ALL) NOPASSWD:/usr/sbin/ipmi-dcmi --get-system-power-statistics" | sudo tee /etc/sudoers.d/green-coding-ipmi-get-machine-energy-stat
-sudo chmod 500 /etc/sudoers.d/green-coding-ipmi-get-machine-energy-stat
-# remove old file name
-sudo rm -f /etc/sudoers.d/ipmi_get_machine_energy_stat
+if [[ $no_ipmi != true ]] ; then
+    print_message "Installing IPMI tools"
+    print_message "Important: If this step fails it means ipmitool is not available on you system"
+    print_message "If you do not plan to use IPMI you can skip the installation by appending '-i'"
+    if lsb_release -is | grep -q "Fedora"; then
+        sudo dnf -y install ipmitool
+    else
+        sudo apt-get install -y freeipmi-tools ipmitool
+    fi
+    print_message "Adding IPMI to sudoers file"
+    echo "ALL ALL=(ALL) NOPASSWD:/usr/sbin/ipmi-dcmi --get-system-power-statistics" | sudo tee /etc/sudoers.d/green-coding-ipmi-get-machine-energy-stat
```

Comment:
This might be a problem as you are opening up the ipmi-dcmi program to non root users which then could access the Data Center Manageability Interface. As you can add more parameters which are not checked

Reply:
no, this is not how sudoers entries work .... see also the other comment. Can you post a link to your understanding of the sudoers entry? Mine is: As soon as you specify a paramters it is locked to exactly this call.